### Business Problem
This is a peer comparison project. Suppose that you are working for Samsung in customer experience management team. The idea is to regularly monitor the customer satisfaction levels and peer company moves. The competitor company is Apple. The objective is to test two main hypothesis.

1. The Samsung Average customer satisfaction score is  minimum 75%.  
2. The overall average satisfaction score of Samsung is same  as Apple. There is no significant difference in the satisfaction  scores 

It might be possible that both hypothesis are correct, one of them is correct or both of them are wrong. Perform the
relevant testing to verify these assumptions. The data is collected for 100 Samsung customers and 100 Apple customers. Their satisfaction scores are recorded. The sample represent the data and it is unbiased.

### Importing Libraries

In [1]:
import numpy as np
import pandas as pd
import scipy.stats as stats

### Importing the Data

In [2]:
testcase = pd.read_csv('Testing_Of_Hypothesis.csv')
testcase.head()

Sr No S_Cust_id  Samsung Score Apple_Cust_id  Apple_Score
0      1   S336499             94       A336525           82
1      2    S73004             99        A73011           83
2      3    S79537             74        A79538           88
3      4   S769168             74       A769222           80
4      5   S609791             83       A609801           74

In [3]:
# shape of the data: 
testcase.shape

(100, 5)

In [4]:
# checking for missing values:
testcase.isnull().sum()

Sr No            0
S_Cust_id        0
Samsung Score    0
Apple_Cust_id    0
Apple_Score      0
dtype: int64

### 1. The Samsung Average customer satisfaction score is minimum 75%.


Testing to check whether the average customer satiscation score for Samsung is atleast 75%.

There is only one sample here: Average customer satisfaction score of Samsung, hence would conduct One Sample T-Test and is one tailed T-Test.

Null Hypothesis and Alternative Hypothesis as:

H0: Samsung Avg Customer Satisfaction score <= 75%\
H1: Samsung Avg Customer Satisfaction score > 75%

Alpha = 5%

In [6]:
Samsung_score = testcase['Samsung Score'].mean()
Samsung_score

84.9

In [7]:
population_mean = Samsung_score*0.75
population_mean 

63.675000000000004

In [8]:
stats.ttest_1samp(a = testcase['Samsung Score'], popmean = 63.675000000000004)

Ttest_1sampResult(statistic=17.91966173190184, pvalue=7.714674273163621e-33)

In [9]:
# Step 1:  Finding the t-score and p-value for the test:

ttest = stats.ttest_1samp(a = testcase['Samsung Score'], popmean = 63.675000000000004)

tscore = round(ttest.statistic,2)
p_value = ttest.pvalue

print("The p-value is {} and T-score is {}.".format(p_value, tscore))


The p-value is 7.714674273163621e-33 and T-score is 17.92.


In [10]:
# Checking if p-value is more than 5% alpha:

p_value > 0.05

False

In [11]:
# Step 2: Checking the values of the sample and population mean:

population_mean = 63.675000000000004

sample_mean = testcase['Samsung Score'].mean()
sample_mean

84.9

#### Conclusion from Step 1: 

p-value =  4.0739777034636007e-13 is less than the alpha of 5%, hence with 95% confidence will reject the Null Hypothesis of the avg customer satisfacton score is minimum 75%.

#### Conclusion from Step 2 :

For us to conclude, whether the average customer satisfaction score is more than 75% for Samsung, we would need to check the values of the sample mean and population mean.

As above, the sample_mean > population_mean, hence the average customer satisfaction score for Samsung is significantly more than 75%.




---------------------------------------------

### 2. The overall average satisfaction score of Samsung is same as Apple. There is no significant difference in the satisfaction scores

Testing to check whether the average customer satiscation score for Samsung is same as that of Apple's.

There is two samples here: Average customer satisfaction score of Samsung and of Apple, hence would conduct Two Sample T-Test and is two tailed T-Test.

Null Hypothesis and Alternative Hypothesis as:

H0: Samsung Avg Customer Satisfaction score == Apple Avg Customer Satisfaction score \
H1: Samsung Avg Customer Satisfaction score != Apple Avg Customer Satisfaction score 
    
    

Alpha = 5%

In [13]:
Apple_score = testcase.Apple_Score.mean()
Apple_score

87.81

In [18]:
Samsung_score = testcase['Samsung Score'].mean()
Samsung_score

84.9

In [21]:
print("The average satisfaction scores of Samsung customers is {}.".format(Samsung_score))
print("The average satisfaction scores of Apple customers is {}.".format(Apple_score))

The average satisfaction scores of Samsung customers is 84.9.
The average satisfaction scores of Apple customers is 87.81.


In [15]:
# Conducting Two Sample Relative t-test

t_test2 = stats.ttest_rel(a = testcase['Apple_Score'], b = testcase['Samsung Score'])

tscore2 = round(t_test2.statistic,2)
p_value2 = round(t_test2.pvalue,4)

print('The p-value is {} and t-score is {}.'.format(p_value2, tscore2))

The p-value is 0.0601 and t-score is 1.9.


#### Conclusion:
    
The p-value of 0.0601 is more than 5% alpha, hence, with 95% confidence we fail to reject Null Hypothesis and conclude there is no signficant difference between the Average Customer Satisifaction score of Samsung and Apple. The overall average customer satisfaction scores of both are same.

---------------------------------------------